sdk/search/search-documents/src/serialization.ts

```diff
@@ -24,7 +24,7 @@ export function deserialize<OutputT>(obj: unknown): OutputT {
 }
 
 function walk(start: unknown, mapper: (val: any) => any): any {
-  const seenMarker = new WeakMap<object, boolean>();
+  const seenMarker = new WeakMap<Record<string, unknown>, boolean>();
```

Comment:
`object` is more correct here since arrays are not considered `Record`s. I think the linter is wrong in this case.

Reply:
Array<T> is a subtype of Record<number, T> I believe.

Reply:
But fwiw I feel like `object` communicates intent better than something like `Record<string | number, unknown>`.

Reply:
For context, the reason `object` is banned by the linter is to not run into this issue: https://github.com/microsoft/TypeScript/issues/21732.

Reply:
I will add the disable line and revert the changes.